### ISSUE TITLE
deprecate and delete twisted.trial.util.findObject

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3108]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3108|
|Type|task|
|Created|2008-03-21 21:47:22Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-findobject-3108|

Nothing uses this, and it's just a crummy rip-off of `namedAny` in the first place.

Attachments:

* [3108.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/250/2504194e61bb6518af0b43095e35d050cc3560bf/3fcbcd76cc3d76d5b1695ef4e4e24aa768361448.diff) (2532 bytes) - added by cyli on 2010-06-15 08:30:51Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3108 3108
type__task task
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_deprecate_findobject_3108 branches/deprecate-findobject-3108
branch_author__jesstess jesstess
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__easy easy
time__1206136042000000 1206136042000000
changetime__1276650285000000 1276650285000000
version__None None
owner__ 
cc__thijs cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_1"></a>|@jml commented|
|-|-|

FWIW, trial.util has long had a strongly-worded warning in the docstring saying that you shouldn't use anything in there.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @cyli<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_4"></a>|@cyli commented|
|-|-|

Deprecated findObject, and added a test for said deprecation.  Also bumped the copyright on util.py.  I notice that there was no copyright for test_util.py, so I added one (as well as a couple simple docstrings) - the oldest commit I saw on that file was from November 2004, so I set the copyright to be 2004-2010.

I also notice there are several deprecated exceptions in util.py, although the deprecation was done via docstring rather than by warnings.  I opened a new ticket ([#4505](https://github.com/twisted/twisted/issues/4505)) to either deprecate them correctly (using warnings) or remove them.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)|@cyli removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_7"></a>|@jesstess commented|
|-|-|

(In [29306]) Branching to 'deprecate-findobject-3108'

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_8"></a>|@jesstess commented|
|-|-|

(In [29307]) Apply patch 3108.diff by cyli.

refs [#3108](https://github.com/twisted/twisted/issues/3108)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_9"></a>|@jesstess set owner to @cyli|
|-|-|

Thanks for working on this cyli! Some feedback:

in utils.py:
 * I think the deprecation warning makes more sense above the function than below it, so you're sure to see it when you scroll down.

in test_utils.py: 
 * I'd probably just say "Tests for L{twisted.trial.util}" instead of  "A collection of tests for L{twisted.trial.util}" for consistency with the other tests.
 * "Test that calling L{findObject} results in a deprecation warning" ==> "Calling L{findObject} results in a deprecation warning.". No need to reaffirm that it's a test.
 * There's some trailing whitespace in test_deprecation that should be removed.

Other than that, [builds](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/deprecate-findobject-3108) look good, and I think this is ready to merge.

I think you have commit bits, so I'm reassigning to you for the merge. If that's true, can you also make branches instead of diffs for your changes in the future, because they're easier to review and test?

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_10"></a>|@cyli commented|
|-|-|

Hmm... sure, I certainly don't mind making branches - someone else had previously told me that for small changes I should make diffs, so I've been using diffs for deprecations/removal of deprecated code.  Perhaps the preferred method should go into the Contributing to Twisted docs.  :)

Thanks for the review!  Going to make changes and merge.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_11"></a>|@jesstess commented|
|-|-|

Interesting -- for me, for anything that isn't purely a documentation change, I want to test it locally and run it through buildbot, which requires a branch, which was my main motivation behind the comment. I agree with you're suggestion; if I can tease out a consensus I'll added it to the wiki.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_12"></a>|@cyli set status to `closed`|
|-|-|

(In [29308]) Merge deprecate-findobject-3108 : deprecate t.trial.util.findObject

Author: cyli,
Reviewer: jesstess
Fixes: [#3108](https://github.com/twisted/twisted/issues/3108)

Deprecate twisted.trial.util.findObject

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
